app/app.py

```diff
@@ -197,10 +224,25 @@ def _create_payload(
         Dict[str, List[Dict[str, Any]]]: the payload to send in the correct format.
     """
     context = messages.get("context", DEFAULT_CONTEXT)
-    input_text = prompt_templates[st.session_state.prompt_template].format(
-        question=messages["prompt_query"], context=context
-    )
+    history: List[Dict[str, str]] = messages.get("history", [])
+    question = messages["prompt_query"]
+
+    if st.session_state.prompt_template == "conversational":
+        if history:
+            history_string = _build_conversation_history_template(history)
+            template = prompt_templates["conversational"]
+            input_text = template.format(
+                history=history_string, question=question, context=context
+            )
+        else:
```

Comment:
1. Why is this `else` block required? What happens if we pass empty history for the first conversation?
2. Why does it use "simple" template as default in `else` clause?

Reply:
This is because the first time when the app is run, there is no history stored in the `st.session_state.nhs_memory`, same for the mind memory. `_build_conversation_history_template()` required some history in order to return a history string. The logic of how history conversation is store and how the string is build could potentially be improved. Open to suggestions.